### ISSUE TITLE
Regression: cannot create validator for vega-lite schema, problem with finding dependencies

### ISSUE THREAD

**Issue by AliciaSchep** [ropensci/jsonvalidate#29]
A schema that is readable by ajv fails to be read in correctly with latest development version of jsonvalidate, choking on the new logic to try to find dependencies.  

Example code to read in:
```r
library(jsonvalidate)
sc <- readr::read_file("https://raw.githubusercontent.com/vegawidget/vegawidget/master/inst/schema/vega-lite/v3.3.0.json")
val <- jsonvalidate::json_validator(sc, engine = "ajv")
```

The error message is:
 Error in context_eval(join(src), private$context) : 
  TypeError: Cannot use 'in' operator to search for '$ref' in null 

And the traceback shows it is coming from a call to `v8$call("find_reference", V8::JS(schema))`

The issue is with the the code for finding the dependencies, but I haven't tracked down the exact cause yet.  This schema works just fine with ajv if passing it in as is without doing the dependency hunting.

**Comment by AliciaSchep** [comment]
I think the issue is that the check for `typeof(x) === "object"` within `find_references` will be true even for `null`, but then the subsequent check to look for `$ref` will error if it is `null`.  Proposed a change in PR #30

**Comment by richfitz** [comment]
Sure, an integration test would be fine for now, and I can dig in to find the underlying reason later on (#28 needs doing and I could do it then)

**Comment by richfitz** [comment]
I found it:

```
        "invalidValues": {
          "description": "Defines how Vega-Lite should handle invalid values (`null` and `NaN`).\n- If set to `\"filter\"` (default), all data items with null values will be skipped (for line, trail, and area marks) or filtered (for other marks).\n- If `null`, all data items are included. In this case, invalid values will be interpreted as zeroes.",
          "enum": [
            "filter",
            null
          ],
          "type": [
            "string",
            "null"
          ]
```

in the vega light schema.  This is likely a bug in the schema as it is not present in the upstream schema

https://github.com/vega/schema/blob/12eee9e8d1ef8227c2e6e2a93d1520c06d014901/vega-lite/v3.3.0.json#L3168-L3177 though oddly it is present in the 3.4.0 schema https://github.com/vega/schema/blob/12eee9e8d1ef8227c2e6e2a93d1520c06d014901/vega-lite/v3.4.0.json#L3127-L3137

I will make the parsing laxer and add a test here now that I know how to trigger it

**Comment by richfitz** [comment]
Fixed in #30
